src/v2/Apps/Fair/Components/ExhibitorsLetterNav.tsx

```diff
@@ -0,0 +1,83 @@
+import React from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import styled from "styled-components"
+import { Text, Flex, Swiper } from "@artsy/palette"
+import { Media } from "v2/Utils/Responsive"
+import { scrollIntoView } from "v2/Utils/scrollHelpers"
+import { ExhibitorsLetterNav_fair } from "v2/__generated__/ExhibitorsLetterNav_fair.graphql"
+
+const LETTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".split("").concat(["0-9"])
+
+interface ExhibitorsLetterNavProps {
+  fair: ExhibitorsLetterNav_fair
+}
+
+export const ExhibitorsLetterNav: React.FC<ExhibitorsLetterNavProps> = ({
+  fair,
+}) => {
+  const letters = fair?.exhibitorsGroupedByName?.map(group => group?.letter)
+
+  const Letters = props => {
+    return (
+      <Flex justifyContent="space-between" {...props}>
+        {LETTERS.map(letter => {
+          const isEnabled = letters?.includes(letter)
+          return (
+            <Letter
+              key={letter}
+              color={isEnabled ? "black100" : "black10"}
+              title={
+                isEnabled ? `View exhibitors starting with “${letter}”` : ""
+              }
+              onClick={() => {
+                if (isEnabled) {
+                  scrollIntoView({
+                    selector: `#jump--letter${letter}`,
+                    offset: 120,
+                    behavior: "smooth",
+                  })
+                }
+              }}
+            >
+              {letter}
+            </Letter>
+          )
+        })}
+      </Flex>
+    )
+  }
+
+  return (
+    <>
+      <Media lessThan="md">
+        <Swiper snap="start">
+          <Letters width={1300} />
```

Comment:
**question**: Is there a way to support the scrolling rail without the fixed width here?

Reply:
Yes, we can use margin for each letter to make space between them